src/AvifLoad.cs

```diff
@@ -26,35 +30,67 @@ public static Document Load(Stream input)
         {
             Document? doc = null;
 
-            using (AvifReader reader = new AvifReader(input, leaveOpen: true))
+            using (IImagingFactory imagingFactory = ImagingFactory.CreateRef())
+            using (AvifReader reader = new AvifReader(input, leaveOpen: true, imagingFactory))
             {
-                Surface? surface = null;
-                bool disposeSurface = true;
-
-                try
+                using (AvifReaderImage image = reader.Decode())
                 {
-                    surface = reader.Decode();
+                    doc = new Document(image.Size);
 
-                    doc = new Document(surface.Width, surface.Height);
+                    BitmapLayer? layer = null;
+                    bool disposeLayer = true;
 
-                    AddAvifMetadataToDocument(doc, reader);
+                    try
+                    {
+                        layer = new(image.Size);
 
-                    doc.Layers.Add(Layer.CreateBackgroundLayer(surface, takeOwnership: true));
-                    disposeSurface = false;
-                }
-                finally
-                {
-                    if (disposeSurface)
+                        PixelFormat pixelFormat = image.WICPixelFormat;
+
+                        if (pixelFormat == PixelFormats.Bgra32)
+                        {
+                            SetOutputLayerDataBgra32(image, layer.Surface);
+                        }
+                        else if (pixelFormat == PixelFormats.Rgba64)
+                        {
+                            SetOutputLayerDataRgba64(image, layer.Surface);
+                        }
+                        else if (pixelFormat == PixelFormats.Rgba128Float)
+                        {
+                            switch (image.HDRFormat)
+                            {
+                                case HDRFormat.PQ:
+                                    SetOutputLayerDataRgba128PQ(doc, image, layer.Surface, imagingFactory);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unsupported HDR format: {image.HDRFormat}.");
+                            }
+                        }
+                        else
+                        {
+                            ExceptionUtil.UnsupportedPixelFormat(pixelFormat);
+                        }
+
+                        doc.Layers.Add(layer);
+                        AddAvifMetadataToDocument(doc, reader, image, imagingFactory);
+                        disposeLayer = false;
+                    }
+                    finally
                     {
-                        surface?.Dispose();
+                        if (disposeLayer)
+                        {
+                            layer?.Dispose();
```

Comment:
I would just let the GC handle things like this. Cleaning up a large managed allocation if there's an exception, I mean. Otherwise I think it's too burdensome to handle every case, IMO.